app/models/concerns/scorecards/filter.rb

```diff
@@ -7,11 +7,12 @@ module Scorecards::Filter
     def self.filter(params = {})
       scope = all
       scope = scope.where("uuid LIKE ?", "%#{params[:uuid].downcase}%") if params[:uuid].present?
-      scope = scope.where("conducted_date >= ?", params[:start_date]) if params[:start_date].present?
+      scope = scope.where("conducted_date BETWEEN ? AND ?", params[:start_date], params[:end_date]) if params[:start_date].present? && params[:end_date].present?
```

Comment:
@hengsokly I think, use planned date is nature then the conducted date and the result match more than,  in e.g: a scorecard planned in 01-Sept-2021, however, the conducted might probably get delay